proto/zitadel/user/v2alpha/user_service.proto

```diff
@@ -254,3 +296,105 @@ message VerifyEmailRequest{
+    }
+  ];
+}
+
+message VerifyPasskeyRegistrationRequest{
+  string user_id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"d654e6ba-70a3-48ef-a95d-37c8d8a7901a\"";
+    }
+  ];
+  string passkey_id = 2 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"fabde5c8-c13f-481d-a90b-5e59a001a076\"";
+    }
+  ];
+  bytes public_key_credential = 3 [
+    (validate.rules).bytes = {min_len: 55, max_len: 1048576},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "JSON representation of public key credential issued by the passkey client";
+      min_length: 55;
+      max_length: 1048576; //1 MB
+    }
+  ];
+  string passkey_name = 4 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"fido key\""
+    }
+  ];
+  string code = 5 [
+    (validate.rules).string = {max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "\"one time code generated by ZITADEL; required to start the passkey registration without user authentication\"";
+      example: "\"SomeSpecialCode\"";
+      max_length: 200;
+    }
+  ];
+}
+
+message VerifyPasskeyRegistrationResponse{
+  zitadel.object.v2alpha.Details details = 1;
+}
+
+message CreatePasskeyRegistrationLinkRequest{
+  string user_id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"d654e6ba-70a3-48ef-a95d-37c8d8a7901a\"";
+    }
+  ];
+  // if no medium is specified, an email is sent with the default url
+  oneof medium {
+    SendPasskeyRegistrationLink send_link = 2;
+    ReturnPasskeyRegistrationCode return_code = 3;
+  }
+}
+
+message CreatePasskeyRegistrationLinkResponse{
+  zitadel.object.v2alpha.Details details = 1;
+  string code = 2;
```

Comment:
code should be `optional` as it is only returned if the `ReturnPasskeyRegistrationCode` is present in the request.
See `verification_code` in `SetEmailResponse`:

https://github.com/zitadel/zitadel/blob/483f88f0cae4e3ab940e63629913555ef887346a/proto/zitadel/user/v2alpha/user_service.proto#L268-L272

Reply:
does it make sense to change it on the request as well as they're only needed if the calling user is permitted directly?